core-plugins/src/main/java/co/cask/hydrator/plugin/batch/source/FileSourceConfig.java

```diff
@@ -137,6 +151,21 @@ protected void validate() {
       throw new IllegalArgumentException("pathField can only be used if inputFormatClass is " +
                                            CombinePathTrackingInputFormat.class.getName());
     }
+    if (format == null || !(format.equalsIgnoreCase("text") ||
+      (format.equalsIgnoreCase("avro")) || format.equalsIgnoreCase("parquet"))) {
+      throw new IllegalArgumentException("Format can only be 'text', 'avro' or 'parquet'");
+    }
+    if (!format.equalsIgnoreCase("text")) {
+      try {
+        new Schema.Parser().parse(inputSchema);
+      } catch (Exception e){
+        throw new IllegalArgumentException("Unable to parse schema with error: " + e.getMessage(), e);
+      }
+    } else if (inputSchema != null && !inputSchema.isEmpty()) {
+      if (!(inputSchema.contains("offset") && inputSchema.contains("body"))) {
```

Comment:
this is wrong, an avro or parquet schema can have a field named 'body'. The validation here should be if format is text, the schema should be equal to the schema used by the code, taking into account whether the pathfield is set or not.